common/spdat.cpp

```diff
@@ -2430,3 +2754,778 @@ bool AegolismStackingIsArmorClassSpell(uint16 spell_id) {
 
 	return 0;
 }
+
+int8 SpellEffectsCount(uint16 spell_id) {
+	if (!IsValidSpell(spell_id)) {
+		return false;
+	}
+
+	int8 x = 0;
+
+	for (int i = 0; i < EFFECT_COUNT; i++) {
+		if (!IsBlankSpellEffect(spell_id, i)) {
+			++x;
+		}
+	}
+
+	return x;
+}
+
+bool IsLichSpell(uint16 spell_id)
+{
+	if (!IsValidSpell(spell_id)) {
+		return false;
+	}
+
+	if (
+		GetSpellTargetType(spell_id) == ST_Self &&
+		IsEffectInSpell(spell_id, SE_CurrentMana) &&
+		IsEffectInSpell(spell_id, SE_CurrentHP) &&
+		spells[spell_id].base_value[GetSpellEffectIndex(spell_id, SE_CurrentMana)] > 0 &&
+		spells[spell_id].base_value[GetSpellEffectIndex(spell_id, SE_CurrentHP)] < 0 &&
+		spells[spell_id].buff_duration > 0
+		) {
+		return true;
+	}
+
+	return false;
+}
+
+bool IsBotSpellTypeDetrimental(uint16 spell_type) {
```

Comment:
All of these bot specific functions should at least live in `spdat_bots.cpp` but share `spdat.h` header and have "Bot" in the function signature.

I'm left wondering why these need to be bot specific when we already have NPC's with spell lists and spell types